gen/redmine.pl

```diff
@@ -0,0 +1,138 @@
+open FILE_PROJECTS,">$fileProjects" or die "Cannot open $fileProjects: $! \n";
+foreach my $projectID (sort keys %$projectStruc) {
+	print FILE_PROJECTS "$projectID, $projectStruc->{$projectID}->{$r_projectName}, $projectStruc->{$projectID}->{$r_description}, $projectStruc->{$projectID}->{$r_isPublic}\n";
+}
+close (FILE_PROJECTS) or die "Cannot close $fileProjects: $! \n";
+
+# USERS: first_name last_name login e-mail eppns
+open FILE_USERS,">$fileUsers" or die "Cannot open $fileUsers: $! \n";
+foreach my $login (sort keys %$userStruc) {
+	print FILE_USERS "$userStruc->{$login}->{$u_firstName}, $userStruc->{$login}->{$u_lastName}, $login, $userStruc->{$login}->{$u_email}, ";
+	print FILE_USERS join( ';', @{$userStruc->{$login}->{$u_eppns}} );
+	print FILE_USERS "\n";
+}
+close (FILE_USERS) or die "Cannot close $fileUsers: $! \n";
+
+# MEMBERSHIPS: project_id description roles
+open FILE_MEMBERSHIPS,">$fileMemberships" or die "Cannot open $fileMemberships: $! \n";
+foreach my $login (sort keys %$membershipStruc){
+	print FILE_MEMBERSHIPS "$membershipStruc->{$login}->{$r_projectID}, $login, ";
+	print FILE_MEMBERSHIPS join(";", @{$membershipStruc->{$login}->{$r_roles}});
+	print FILE_MEMBERSHIPS "\n";
+}
+close (FILE_MEMBERSHIPS) or die "Cannot close $fileMemberships: $! \n";
+
+perunServicesInit::finalize;
+
+
+##############################################################################
+#   Only subs definitions down there
+##############################################################################
+## creates structure for projects.csv file
+sub processProjects {
+	my $project = shift;
+
+	my %resourceAttributes = attributesToHash $project->getAttributes;
+
+	my $projectID =		$resourceAttributes{$A_RESOURCE_PROJECTID};
+	my $projectName =	$resourceAttributes{$A_RESOURCE_PROJECTNAME};
+	my $description =	$resourceAttributes{$A_RESOURCE_DESCRIPTION};
+	my $isPublic =		"True";
+	my @roles =		$resourceAttributes{$A_RESOURCE_ROLE};
+
+	if(exists $projectStruc->{$projectID} && $projectStruc->{$projectID}->{$r_projectName} != $projectName){
+		die "Cannot exists more projects with same projectID and with different name! \n";
+	}
+
+	$projectStruc->{$projectID}->{$r_projectName} = $projectName;
+	$projectStruc->{$projectID}->{$r_description} = $description;
+	$projectStruc->{$projectID}->{$r_isPublic}    = $isPublic;
+
+		foreach my $memberData ($project->getChildElements) {
+			processUsers $projectID, $memberData ,@roles;
+		}
+}
+
+## creates structure for users.csv file
+sub processUsers {
+	my ($projectID, $memberData ,@roles) = @_;
+
+	my %memberAttributes = attributesToHash $memberData->getAttributes;
+	my $firstName = $memberAttributes{$A_USER_FIRSTNAME};
+	my $lastName = $memberAttributes{$A_USER_LASTNAME};
+	my $login = $memberAttributes{$A_USER_LOGIN};
+	my $email = $memberAttributes{$A_USER_EMAIL};
+	my @eppns = @{$memberAttributes{$A_USER_EPPNS}};
+
+	$userStruc->{$login}->{$u_firstName} = $firstName;

```

Comment:
It's better to use strings as key for hash. At least it can help during debug. 
So, you can use something like following:
```perl
$userStruc->{$login}->{$A_USER_FIRSTNAME} = $firstName;
```

or even better, you can replace all code in processUsers subroutine with something like following. It might put into $userStruc more attribute than you want, but ussualy it doesn't matter.
```perl 
sub processUsers {
  my ($projectID, $memberData ,@roles) = @_;
  my %memberAttributes = attributesToHash $memberData->getAttributes;

  $userStruc->{$memberAttributes{$A_USER_LOGIN}} = $%memberAttributes unless defined $userStruc->{$memberAttributes{$A_USER_LOGIN}};

  processMemberships $login, $projectID, @roles;
}
```